### PR TITLE
picasso composable centauri clients build and runs

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
**Picasso Composable hyperspace runs and DOT transfered**

- updated centauri to version without beefy
- hyperspace image and binary with patched clients of picasso and composvble
- updated relay runtime (allowed to use exact version composable depends on)
- deleted dali runtime from hyperspace
- made chains explicit in devnet configs to prevent config errors
- tuned debug prameters for hyperspace and devnet rust builds
- split out substrate rust build parameters more

### CODE REVIEW

File: devnets/specs/centauri.nix [hunk 1]
```diff
@@ -1,123 +1,129 @@
...
+                  "transfer"
+                  "--version"
+                  "ics20-1"
+                  "--order"
+                  "unordered"
+                ];
+                inherit configPathSource configPathContainer pkgs packages
+                  devnetTools;
+                dependsOn = dependsOnCreateConnection;
+                restartPolicy = "no";
```
Comment (blasrodri) [inline devnets/specs/centauri.nix:111]: why did you change the restart policy?

File: devnets/specs/centauri.nix [hunk 2]
```diff
@@ -1,123 +1,129 @@
...
+                  "--port-id"
+                  "transfer"
+                  "--version"
+                  "ics20-1"
+                  "--order"
+                  "unordered"
+                ];
+                inherit configPathSource configPathContainer pkgs packages
+                  devnetTools;
+                dependsOn = dependsOnCreateConnection;
```
Comment (blasrodri) [inline devnets/specs/centauri.nix:110]:
```suggestion
                dependsOn = dependsOnCreateChannels;
```

**Review by blasrodri — APPROVED** [review]

File: devnets/all.nix [hunk 3]
```diff
@@ -13,18 +13,18 @@
...
+          devnet-b = packages.zombienet-picasso-centauri-b;
         });
 
       devnet-centauri-no-relay =
         pkgs.composable.mkDevnetProgram "devnet-centauri"
         (import ./specs/centauri.nix {
           inherit pkgs devnetTools packages;
           devnet-a = packages.zombienet-picasso-centauri-a;
-          devnet-b = packages.zombienet-composable-centauri-b;
+          devnet-b = packages.zombienet-picasso-centauri-b;
```
Comment (blasrodri) [inline devnets/all.nix:27]: do you want to merge this? Picasso<>Picasso

File: inputs/paritytech/polkadot.nix [hunk 4]
```diff
@@ -1,35 +1,72 @@
...
+          name = "rococo-wasm-runtime";
+          dontUnpack = true;
+          src = pkgs.fetchurl {
+            url =
+              "https://github.com/paritytech/polkadot/releases/download/v0.9.36/rococo_runtime-v9360.compact.compressed.wasm";
+            hash = "sha256-inq526PxU2f4+m4RSTiv5oOpfSZfnQpXkhpYmqZ9gOs=";
+          };
+          installPhase = ''
+            mkdir -p $out/lib
+            cp $src $out/lib/rococo_runtime.compact.compressed.wasm
```
Comment (andresmechali) [inline inputs/paritytech/polkadot.nix:46]: shouldn't this be `rococo_runtime-v9360.compact.compressed.wasm`?

File: inputs/paritytech/zombienet/default.nix [hunk 5]
```diff
@@ -29,7 +29,7 @@ in with prelude; rec {
       env = [{
         name = "RUST_LOG";
         value =
-          "info,runtime=debug,parachain=trace,cumulus-collator=trace,aura=trace,xcm=trace,pallet_ibc=trace"
+          "info,runtime=debug,parachain=trace,cumulus-collator=trace,aura=trace,xcm=trace,pallet_ibc=trace,hyperspace=trace,hyperspace_parachain=trace,ics=trace,ics::routing=trace,ics::channel=trace"
```
Comment (blasrodri) [inline inputs/paritytech/zombienet/default.nix:32]: do you want to merge with all logs on trace?

**Review by blasrodri — APPROVED** [review]

**Review by andresmechali — APPROVED** [review]

File: code/parachain/runtime/primitives/src/topology.rs [hunk 6]
```diff
@@ -154,8 +155,25 @@ impl Composable {
...
+			AssetInfo {
+				name: Some(
+					BiBoundedAssetName::from_vec(b"Polkadot".to_vec())
+						.expect("String is within bounds"),
+				),
+				symbol: Some(
+					BiBoundedAssetSymbol::from_vec(b"DOT".to_vec())
+						.expect("String is within bounds"),
+				),
+				decimals: Some(10),
```
Comment (kkast) [inline code/parachain/runtime/primitives/src/topology.rs:170]: in the migrated dot we set decimals to 12

File: code/parachain/runtime/primitives/src/topology.rs [hunk 7]
```diff
@@ -154,8 +155,25 @@ impl Composable {
...
+				existential_deposit: 1_000_000,
+				ratio: Some(rational!(2143 / 100_000_000)),
```
Comment (kkast) [inline code/parachain/runtime/primitives/src/topology.rs:172]: this is also different from migrated dot on composable 
existential_deposit: 214_300_000,
ratio: Some(rational!(2143 / 1_000_000))

File: code/parachain/runtime/primitives/src/topology.rs [hunk 8]
```diff
@@ -132,10 +133,10 @@ impl Composable {
 				),
 				decimals: Some(6),
 				existential_deposit: 10_000,
-				ratio: Some(rational!(375 / 1_000_000)),
+				ratio: Some(rational!(375 / 1_000_000_000)),
```
Comment (kkast) [inline code/parachain/runtime/primitives/src/topology.rs:136]: in picasso usdt id is 130 and parameters:
existential_deposit: 1500,
ratio: Some(rational!(15 / 1_000_000_000)),

File: code/parachain/runtime/primitives/src/topology.rs [hunk 9]
```diff
@@ -92,9 +93,9 @@ impl Picasso {
 					BiBoundedAssetSymbol::from_vec(b"DOT".to_vec())
 						.expect("String is within bounds"),
 				),
-				decimals: Some(12),
-				existential_deposit: 1_000_000_000,
-				ratio: Some(rational!(375 / 1_000)),
+				decimals: Some(10),
```
Comment (kkast) [inline code/parachain/runtime/primitives/src/topology.rs:96]: decimals: Some(12),
existential_deposit: 214_300_000,

File: code/parachain/runtime/primitives/src/topology.rs [hunk 10]
```diff
@@ -75,11 +75,12 @@ impl Picasso {
 				),
 				decimals: Some(6),
 				existential_deposit: 10_000,
-				ratio: Some(rational!(375 / 1_000_000)),
+				ratio: Some(rational!(15 / 1_000_000_000)),
```
[the remark this answers is not in the document]
Comment (kkast) [reply code/parachain/runtime/primitives/src/topology.rs:78]: in picasso usdt id is 130 and parameters:
existential_deposit: 1500,
ratio: Some(rational!(15 / 1_000_000_000)),

**Review by kkast — CHANGES_REQUESTED** [review]

**Review by kkast — APPROVED** [review]

File: code/Cargo.toml [hunk 11]
```diff
@@ -72,12 +72,12 @@ rpath = false
 
 
 [workspace.dependencies]
-ibc = { git = "https://github.com/ComposableFi/centauri", rev = "c8aa82d95c817024f814c76131dba99c305fa1c4", default-features = false }
-ibc-rpc = { git = "https://github.com/ComposableFi/centauri", rev = "c8aa82d95c817024f814c76131dba99c305fa1c4", default-features = false }
-ibc-primitives = { git = "https://github.com/ComposableFi/centauri", rev = "c8aa82d95c817024f814c76131dba99c305fa1c4", default-features = false }
-ibc-runtime-api = { git = "https://github.com/ComposableFi/centauri", rev = "c8aa82d95c817024f814c76131dba99c305fa1c4", default-features = false }
-pallet-ibc = { git = "https://github.com/ComposableFi/centauri", rev = "c8aa82d95c817024f814c76131dba99c305fa1c4", default-features = false }
-pallet-ibc-ping = { git = "https://github.com/ComposableFi/centauri", rev = "c8aa82d95c817024f814c76131dba99c305fa1c4", default-features = false }
+ibc = { git = "https://github.com/dzmitry-lahoda-forks/centauri", rev = "15b72401a8292f248c2f3afe1e29cf83bf359c36", default-features = false }
```
Comment (blasrodri) [inline code/Cargo.toml:75]: why not use Centauri?

**Pull request merged**